### PR TITLE
restructure the monitoring to automate the monitoring 

### PR DESCRIPTION
**Pull request by harshad16**
restructure the monitoring to automate the monitoring 
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

## Related Issues and Dependencies

Related-to: https://github.com/thoth-station/thoth-application/issues/854

### CODE REVIEW

File: monitoring/overlays/cnv-prod/alerting-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+          # Alert for amcheck database corruption check.
+        - alert: ThothDatabaseCorruptionCheckProd
+          expr: thoth_graphdb_is_corrupted{instance="metrics-exporter-thoth-infra-prod.apps.zero.massopen.cloud:80"} == 1
+          for: 5m
+          annotations:
+            summary: "Database schema is corrupted, all services need to be stopped."
+            severity: "HIGH"
+
+          # Alert for alembic version mismatch between components and database.
```
Comment (pacospace) [inline monitoring/overlays/cnv-prod/alerting-rules.yaml:134]: # Alert for alembic version table corruption

File: monitoring/overlays/cnv-prod/alerting-rules.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        - alert: ThothMultipleAlembicRowsProd
+          expr: thoth_graphdb_alembic_table_check{instance="metrics-exporter-thoth-infra-prod.apps.zero.massopen.cloud:80"} == 1
+          for: 5m
+          annotations:
+            summary: "Database table for alembic version has more than one row, database is corrupted, all services need to be stopeed."
```
Comment (pacospace) [inline monitoring/overlays/cnv-prod/alerting-rules.yaml:139]: stopped

File: monitoring/base/alertmanager.yaml [hunk 3]
```diff
@@ -28,6 +28,6 @@ kind: Alertmanager
 metadata:
   name: thoth-monitoring
 spec:
-  externalUrl: "http://alert-manager-thoth-infra-prod.apps.zero.massopen.cloud/"
+  externalUrl: ""
```
Comment (pacospace) [inline monitoring/base/alertmanager.yaml:31]: why this is empty?

**Review by pacospace — COMMENTED** [review]
Few minor comments, otherwise 💯

File: monitoring/overlays/cnv-prod/alerting-rules.yaml (hunk 1 above)
Comment (harshad16) [reply monitoring/overlays/cnv-prod/alerting-rules.yaml:134]: Thanks :+1:

File: monitoring/overlays/cnv-prod/alerting-rules.yaml (hunk 2 above)
Comment (harshad16) [reply monitoring/overlays/cnv-prod/alerting-rules.yaml]: Thanks :+1:

File: monitoring/base/alertmanager.yaml (hunk 3 above)
Comment (harshad16) [reply monitoring/base/alertmanager.yaml:31]: we wanted to keep the `base` contain separate from direct cluster-related values, so the value is embedded in the overlay.
https://github.com/thoth-station/thoth-application/blob/a0e1d0fcaef73ce8b72e3d77c2ffd3255bc2ab7e/monitoring/overlays/cnv-prod/kustomization.yaml#L20

**Review by pacospace — APPROVED** [review]
🚀 Thanks! lgtm!

**Pull request merged**